[quant] Add quantized Sigmoid module

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #45920 [quant][fx][graphmode][api] Change API for custom module
* #45919 [quant][fx][graphmode] Run symbolic_trace in quantization
* #45885 [quant][fx][graphmode] Add support for sigmoid
* #45884 [quant][eagermode] Add support for sigmoid module
* **#45883 [quant] Add quantized Sigmoid module**
* #45882 [quant] Add quantized::sigmoid that take output_scale/output_zero_point as input

Summary:

Test Plan:
python test/test_quantization.py TestStaticQuantizedModule.test_sigmoid

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D24129116](https://our.internmc.facebook.com/intern/diff/D24129116)

jerryzh168 committed:
[quant] Add quantized Sigmoid module

Summary:

Test Plan:
python test/test_quantization.py TestStaticQuantizedModule.test_sigmoid

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

z-a-f commented on torch/nn/quantized/modules/activation.py:
```diff
@@ -149,3 +149,24 @@ def _get_name(self):
+        zero_point: quantization zero point of the output tensor
+    """
+
+    def __init__(self, output_scale: float, output_zero_point: int):
```
Can we have default args? It is safe to assume those for this function

jerryzh168 replied:
not sure, it's not likely that user is going to construct quantized model from quantized modules from scratch anyways, what is the benefit of having default args?

jerryzh168 committed:
Update on "[quant] Add quantized Sigmoid module"


Summary:

Test Plan:
python test/test_quantization.py TestStaticQuantizedModule.test_sigmoid

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D24129116](https://our.internmc.facebook.com/intern/diff/D24129116)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] Add quantized Sigmoid module"


Summary:

Test Plan:
python test/test_quantization.py TestStaticQuantizedModule.test_sigmoid

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D24129116](https://our.internmc.facebook.com/intern/diff/D24129116)

[ghstack-poisoned]

raghuramank100 commented on torch/nn/quantized/modules/activation.py:
```diff
@@ -149,3 +149,24 @@ def _get_name(self):
         return cls(float(scale), int(zero_point), mod.negative_slope, mod.inplace)
+
+class Sigmoid(torch.nn.Sigmoid):
+    r"""This is the quantized equivalent of :class:`~torch.nn.LeakyReLU`.
```
The docs are off here, can we refer to sigmoid instead of leakyRelu?

jerryzh168 replied:
oh sorry, will fix in next PR
